src/django_browser_reload/views.py

```diff
@@ -145,19 +150,40 @@ def events(request: HttpRequest) -> HttpResponseBase:
     if not request.accepts("text/event-stream"):
         return HttpResponse(status=HTTPStatus.NOT_ACCEPTABLE)
 
-    def event_stream() -> Generator[bytes, None, None]:
-        while True:
-            yield message("ping", versionId=version_id)
-
-            should_reload = should_reload_event.wait(timeout=PING_DELAY)
-            if should_reload:
-                should_reload_event.clear()
-                yield message("reload")
-
-    response = StreamingHttpResponse(
-        event_stream(),
-        content_type="text/event-stream",
-    )
+    if isinstance(request, ASGIRequest):
+        if DJANGO_VERSION < (4, 2):
+            err_msg = "We cannot support hot reload with ASGI for Django < 4.2"
+            warnings.warn(err_msg, stacklevel=2)
+            return HttpResponse(status=HTTPStatus.NOT_ACCEPTABLE)
+
+        async def async_event_stream() -> AsyncGenerator[bytes, None]:
+            while True:
+                await asyncio.sleep(PING_DELAY)
+                yield message("ping", versionId=version_id)
+
+                if should_reload_event.is_set():
+                    should_reload_event.clear()
+                    yield message("reload")
+
+        response = StreamingHttpResponse(
+            async_event_stream(),
+            content_type="text/event-stream",
+        )
+    else:
+
+        def event_stream() -> Generator[bytes, None, None]:
+            while True:
+                yield message("ping", versionId=version_id)
+
+                should_reload = should_reload_event.wait(timeout=PING_DELAY)
+                if should_reload:
+                    should_reload_event.clear()
+                    yield message("reload")
+
+        response = StreamingHttpResponse(
+            event_stream(),
+            content_type="text/event-stream",
+        )
```

Comment:
can be extracted to common branch by using `event_stream` as the name for the async generator too